src/main/java/com/bc/zarr/storage/OmeroFileSystemStore.java

```diff
@@ -0,0 +1,71 @@
+package com.bc.zarr.storage;
+
+import com.amazonaws.services.s3.model.AmazonS3Exception;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.FileSystem;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+/**
+ * Overridden FileSystemStore. Implemented to catch exceptions
+ * when missing chunks are encountered rather than allowing them to propogate
+ * causing errors.
+ */
+public class OmeroFileSystemStore extends FileSystemStore {
+
+    private Path omeroInternalRoot; // Field in FileSystemStore is private
+    
+    /**
+     * Constructor.
+     *
+     * @param path The path to the zarr root
+     * @param fileSystem The FileSystem used to access the file
+     */
+    public OmeroFileSystemStore(String path, FileSystem fileSystem) {
+        super(path, fileSystem);
+        if (fileSystem == null) {
+            omeroInternalRoot = Paths.get(path);
+        } else {
+            omeroInternalRoot = fileSystem.getPath(path);
+        }
+    }
+
+    /**
+     * Constructor.
+     *
+     * @param rootPath Path to the zarr root
+     */
+    public OmeroFileSystemStore(Path rootPath) {
+        super(rootPath);
+        omeroInternalRoot = rootPath;
+    }
+    
+    /**
+     * Gets an input stream for the file contents if it exists and is readable,
```

Comment:
Either this Javadoc or the implementation below could also benefit from a comment explaining the special handling of AmazonS3Exception with 404 error codes

Reply:
Agreed.